compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3544,9 +3544,38 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
         arrayTypeNode.elemtype = createTypeNode(arrayTypeDescriptorNode.memberTypeDesc());
         arrayTypeNode.dimensions = dimensionSize;
         arrayTypeNode.sizes = sizes;
+        for (BLangExpression member : sizes) {
+            if (member.getKind() == NodeKind.LITERAL &&
+                    ((BLangLiteral) member).value.equals(INFERRED_ARRAY_INDICATOR)) {
+                if (setIsInferableFlag(arrayTypeDescriptorNode)) {
+                    arrayTypeNode.inferredArrayValidateState = 1;
+                } else {
+                    arrayTypeNode.inferredArrayValidateState = -1;
+                }
+                break;
+            }
+        }
         return arrayTypeNode;
     }
 
+    private boolean setIsInferableFlag(Node node) {
+        if (!(node.kind() == SyntaxKind.TYPED_BINDING_PATTERN) && node.parent() != null) {
+            return setIsInferableFlag(node.parent());
+        }
+        if (node.parent() == null) {
+            return false;
+        }
```

Comment:
Can't we do this null check at the begining of method. If yes we can simplify previous check too i.e. remove `node.parent() != null`

Reply:
Removed this logic from the BLangNodeBuilder in https://github.com/ballerina-platform/ballerina-lang/pull/34612/commits/dd4a64875eca77bcc52d1ceb139a8f36621b4cea